qiime2/core/util.py

```diff
@@ -573,45 +572,51 @@ def gpg_find_key(key_selector):
     for line in output.splitlines():
         parts = line.split(':')
         tag = parts[0]
+
         # public key tag; the primary key info that matches
-        # the given uid or fingerprint will be here
+        # the fingerprint will be here
         if tag == 'pub':
             in_primary = True
-            length = parts[2] or '0'
-            algorithm_num = parts[3] or ''
-            curve = parts[15] if len(parts) >= 16 and parts[15] else None
-            key_info['length'] = int(length) if length.isdigit() else 0
+            length = parts[2] if len(parts) > 2 else '0'
+            algorithm_num = parts[3] if len(parts) > 3 else ''
+            curve = parts[15] if len(parts) > 15 and parts[15] else None
```

Comment:
why the `and parts[15]` on this line when the others don't check the truthiness of the value at their index?